client/admin/matchmaking-times.jsx

```diff
@@ -0,0 +1,386 @@
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import CheckBox from '../material/check-box.jsx'
+import FlatButton from '../material/flat-button.jsx'
+import LoadingIndicator from '../progress/dots.jsx'
+import RaisedButton from '../material/raised-button.jsx'
+import { ScrollableContent } from '../material/scroll-bar.jsx'
+import Tabs, { TabItem } from '../material/tabs.jsx'
+
+import { MATCHMAKING_TYPE_1V1 } from '../../common/constants'
+
+import CheckIcon from '../icons/material/baseline-check_circle-24px.svg'
+
+import {
+  getMatchmakingTimesHistory,
+  getMatchmakingTimesFuture,
+  getMatchmakingTimesPast,
+  addMatchmakingTime,
+  deleteMatchmakingTime,
+} from './action-creators'
+
+import { colorTextSecondary, colorError, colorSuccess, amberA400 } from '../styles/colors'
+import { Body1, Subheading } from '../styles/typography'
+
+const MATCHMAKING_TIMES_LIMIT = 10
+
+const dateFormat = new Intl.DateTimeFormat(navigator.language, {
+  year: 'numeric',
+  month: 'short',
+  day: '2-digit',
+  hour: 'numeric',
+  minute: '2-digit',
+  timeZoneName: 'short',
+})
+
+const LoadingArea = styled.div`
+  margin: 16px 0;
+`
+
+const ErrorText = styled(Subheading)`
+  color: ${colorError};
+`
+
+const HistoryContainer = styled.table`
+  width: 100%;
+
+  th,
+  td {
+    border: none;
+    padding: 5px 0px;
+  }
+
+  th {
+    color: ${colorTextSecondary};
+    text-align: left;
+    font-weight: 500;
+  }
+
+  td {
+    height: 64px;
+    vertical-align: center;
+  }
+
+  th:first-child {
+    width: 50%;
+  }
+
+  th:last-child {
+    padding-left: 16px;
+  }
+`
+
+const EnabledText = styled.span`
+  color: ${colorSuccess};
+`
+
+const DisabledText = styled.span`
+  color: ${colorError};
+`
+
+const CurrentText = styled.span`
+  color: ${amberA400};
+`
+
+const FinishedText = styled.span`
+  color: ${colorTextSecondary};
+`
+
+class MatchmakingTimesHistory extends React.PureComponent {
+  static propTypes = {
+    history: PropTypes.object,
+    futureTimesPage: PropTypes.number.isRequired,
+    pastTimesPage: PropTypes.number.isRequired,
+    onLoadMoreFuture: PropTypes.func.isRequired,
+    onLoadMorePast: PropTypes.func.isRequired,
+    onDelete: PropTypes.func.isRequired,
+  }
+
+  render() {
+    const {
+      history,
+      futureTimesPage,
+      pastTimesPage,
+      onLoadMoreFuture,
+      onLoadMorePast,
+      onDelete,
+    } = this.props
+
+    if (!history) return null
+
+    if (history.isRequesting) {
+      return (
+        <LoadingArea>
+          <LoadingIndicator />
+        </LoadingArea>
+      )
+    }
+
+    if (history.lastError) {
+      return (
+        <>
+          <p>
+            Something went wrong while trying to retrieve the matchmaking times history. The error
+            message was:
+          </p>
+          <ErrorText as='p'>{history.lastError.message}</ErrorText>
+        </>
+      )
+    }
+
+    if (history.sortedList.isEmpty()) {
```

Comment:
For a demonstrations as to why you shouldn't put expensive stuff in getters, look at how many times you use `history.sortedList` in this function. It doesn't *feel* expensive, it just looks like a variable access. But every access here is creating an entirely new list, sorting it, then reversing it. And that happens even if you're just checking if there's at least one entry. If it had been a function you probably would have used it once and stored the value, then reused that value throughout the function.